sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mlme.c

```diff
@@ -310,17 +349,21 @@ void gnrc_lorawan_process_fopts(gnrc_lorawan_t *mac, uint8_t *fopts,
 
     void (*cb)(gnrc_lorawan_t *, uint8_t *p) = NULL;
 
-    for (uint8_t pos = 0; pos < size; pos += ret) {
+    for (uint8_t pos = 0; pos < size; pos = ret) {
         switch (fopts[pos]) {
             case GNRC_LORAWAN_CID_LINK_CHECK_ANS:
                 ret += GNRC_LORAWAN_FOPT_LINK_CHECK_ANS_SIZE;
```

Comment:
here_
```suggestion
                ret = GNRC_LORAWAN_FOPT_LINK_CHECK_ANS_SIZE;
```

Same below